quality/extract_cci_approved_executed.py

```diff
@@ -13,14 +13,17 @@ def get_cci_metadata(prune=False):
     countries = spark.table('indicator.country').toPandas()
     
     for filename in tqdm(files):
-        try:
-            for sheet_name in ['Approved', 'Executed']: # quick & dirty way to make sure both sheets exit
-                df = pd.read_excel(filename, sheet_name=sheet_name, na_values=['..'])
-        except ValueError as e:
-            print(f"Error reading {sheet_name} from {filename}")
-            print(e)
+        xls = pd.ExcelFile(filename)
+        if 'Executed' not in xls.sheet_names and 'Approved' not in xls.sheet_names:
+            print(f"Neither 'Executed' nor 'Approved' sheet found in {filename}")
             continue
-        
+
+        sheet_name = 'Executed'
+        if sheet_name not in xls.sheet_names:
+            sheet_name = 'Approved'
+
+        df = pd.read_excel(filename, sheet_name=sheet_name, na_values=['..'])
```

Comment:
Since we already have the ExcelFile object at the line 16,  just use xls.parse(...) or read_excel(xls)?

Reply:
Good spotting! Done.